File: Programmers/Level2/42578.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 0;
    unordered_map<string, int> map;
    
    for (vector<string> cloth : clothes) {
        // clothes.push_back();
        auto iter = map.find(cloth[1]);
        if (iter == map.end()) {
            map.insert({ cloth[1], 1 });
        }
        else {
            map[cloth[1]] = map[cloth[1]] + 1;
        }
    }
    
    int multiplier = 1;
    for (auto cloth : map) {
        // 안 입는 경우의 수 추가
        multiplier *= (cloth.second + 1);
    }
    
    // 모두 안 입는 경우 제외
    answer += (multiplier - 1);

    return answer;
}

File: Programmers/Level2/42578.js
function solution(clothes) {
  const cases = {};

  clothes.map((clothObj) => {
    const cloth = clothObj[1];

    if (cases[cloth]) ++cases[cloth];
    else cases[cloth] = 1;
  });

  const caseCnts = Object.keys(cases).map((key) => cases[key]);

  return caseCnts.reduce((accum, caseCnt) => accum * (caseCnt + 1), 1) - 1;
}
